sdk/key_vault/src/key.rs

```diff
@@ -0,0 +1,467 @@
+    key_id: String,
+}
+
+/// The signing/verification algorithm identifier
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum SignatureAlgorithm {
+    ES256,  // ECDSA using P-256 and SHA-256, as described in https://tools.ietf.org/html/rfc7518.
+    ES256K, // ECDSA using P-256K and SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    ES384,  // ECDSA using P-384 and SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    ES512,  // ECDSA using P-521 and SHA-512, as described in https://tools.ietf.org/html/rfc7518
+    PS256, // RSASSA-PSS using SHA-256 and MGF1 with SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    PS384, // RSASSA-PSS using SHA-384 and MGF1 with SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    PS512, // RSASSA-PSS using SHA-512 and MGF1 with SHA-512, as described in https://tools.ietf.org/html/rfc7518
+    RS256, // RSASSA-PKCS1-v1_5 using SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    RS384, // RSASSA-PKCS1-v1_5 using SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    RS512, // RSASSA-PKCS1-v1_5 using SHA-512, as described in https://tools.ietf.org/html/rfc7518
+    Custom(String),
+}
+
+impl Default for SignatureAlgorithm {
+    fn default() -> Self {
+        SignatureAlgorithm::Custom("Not Set".to_string())
+    }
+}
+
+impl Display for SignatureAlgorithm {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl<'a, T: TokenCredential> KeyClient<'a, T> {
+    /// Gets the public part of a stored key.
+    /// The get key operation is applicable to all key types.
+    /// If the requested key is symmetric, then no key material is released in the response.
+    /// This operation requires the keys/get permission.
+    ///
+    /// GET {vaultBaseUrl}/keys/{key-name}/{key-version}?api-version=7.1
+    pub async fn get_key(
+        &mut self,
+        key_name: &str,
+        key_version: Option<&str>,
+    ) -> Result<KeyVaultKey, KeyVaultError> {
+        let mut uri = format!("{}/keys/{}", self.keyvault_endpoint, key_name);
+        if let Some(ver) = key_version {
+            uri.push('/');
+            uri.push_str(ver);
+        }
+        let uri = Url::parse_with_params(&uri, &[("api-version", API_VERSION)]).unwrap();
+
+        let resp_body = self.get_authed(uri.to_string()).await?;
+        let response = serde_json::from_str::<KeyVaultKey>(&resp_body)?;
+        Ok(response)
+    }
+
+    /// Creates a signature from a digest using the specified key.
+    /// The SIGN operation is applicable to asymmetric and symmetric keys stored in Azure Key Vault since this operation uses the private portion of the key.
+    /// This operation requires the keys/sign permission.
+    pub async fn sign(
+        &mut self,
+        key_name: &str,
+        key_version: &str,
+        digest: &str,
+        algorithm: SignatureAlgorithm,
```

Comment:
Algorithm should be listed first, though I wouldn't hold the PR up for this since I opened https://github.com/Azure/azure-sdk-for-rust/issues/197 to track moving this to `CryptographyClient`. That said, it'd be a minor change, I think, to do it now since some other small changes are needed.